pkg/component/runtime/runtime_comm.go

```diff
@@ -141,30 +137,65 @@ func (c *runtimeComm) CheckinExpected(expected *proto.CheckinExpected) {
 	} else {
 		expected.AgentInfo = nil
 	}
-
-	// Lock to avoid race if this function is called from the different go routines
-	c.checkinExpectedLock.Lock()
-
-	// Empty the channel
-	c.ClearPendingCheckinExpected()
-
-	// Put the new expected state in
 	c.checkinExpected <- expected
-
-	c.checkinExpectedLock.Unlock()
-}
-
-func (c *runtimeComm) ClearPendingCheckinExpected() {
-	select {
-	case <-c.checkinExpected:
-	default:
-	}
 }
 
 func (c *runtimeComm) CheckinObserved() <-chan *proto.CheckinObserved {
 	return c.checkinObserved
 }
 
+// latestCheckinExpected ensures that the latest expected checkin is used
+func (c *runtimeComm) latestCheckinExpected(exp *proto.CheckinExpected) *proto.CheckinExpected {
```

Comment:
This function could be unit tested by itself it looks like. Do we exercise any of the edge cases already? If not it is probably worth adding some sanity tests.

Reply:
Being that I have changed the implementation I was able to remove all this extra code. It is no longer required.